### DISCUSSION TITLE
My workflow on FOSDEM2021

### DISCUSSION THREAD

**Discussion by steelman** [EventFahrplan/EventFahrplan#379] [Show and tell]
First things first: the app is great! It helped me a lot during the last FOSDEM. However, my workflow is not entirely supported. Let me share it, so you can decide if it makes sense to support it.

I start with browsing the schedule around to find interesting lectures. From left to right from, top to bottom and back again. At this point I don't mind if the events overlap in time. I mark all events that seem interesting as favorite. This step works fine.

Then I go to favorites and make the first step for holes in my lineup.

**Issue**: events have their starting time and duration displayed but not ending time so I need to do basic math if two adjacent events overlap. Even more math is needed if they don't overlap and I want to know how much time is there in between.

After identifying hole I go back to the schedule and look for more interesting events in the time holes I've got.

**Issue**: I miss some horizontal guides I could set in the schedule view to mark a period I would like to fill with events.

Rinse and repeat.

When the favorites list is full or more I select events I want to attend. I set alarms on them.

**Issue**: The bell that marks event on the schedule screen isn't there on the favorites list and I can't see which event I want to attend by looking at the favorites list.

When the time comes, the alarm rings, I go to the event.

**Issue**: There is no way to mark an event I attended, so after a conference, when I am going to watch videos I can't distinguish between those which I wanted to see and those I actually saw.

This 3 marks workflow is even more important when attending a conference in person. Last year I had my favorites list (in a different app) overflowing with backup events in case a room was full. In online mode it also makes sense to have a backup plan when a lecture appears boring or streaming breaks (fortunately it didn't)

**Comment by johnjohndoe** [comment]
Thank you a lot for your feedback. I enjoyed reading it and consider it as very valuable for the development of the app. I hope more users take your post as an example to follow after. Also, I am glad that the app helped to make FOSDEM a nice experience.

> I mark all events that seem interesting as favorite. This step works fine.
> 
> Then I go to favorites and make the first step for holes in my lineup.

I understand that you need the favorites screen to spot the holes if the conference program is so big that the schedule grid does not help that. Here a zoom feature would be a solution.

> **Issue**: events have their starting time and duration displayed but not ending time ...

Yes, please join the conversation in #227 and #231.

> **Issue**: I miss some horizontal guides I could set in the schedule view to mark a period I would like to fill with events.

Interesting. I never thought about this. Guides would be useful for the discovery phase and could be removed later. Alternatively, a zoom feature or to hide columns (see #100) could help here.

> When the favorites list is full or more I select events I want to attend. I set alarms on them.

If I understand correctly, you are using alarms as a secondary step to filter your favorites list. I never thought about it this way.

> **Issue**: The bell that marks event on the schedule screen isn't there on the favorites list and I can't see which event I want to attend by looking at the favorites list.

It might be a good idea to merge the alarms with the favorites screen. I never thought about this. One challenge here would be to come up with a good design to fit in all information for a session.

> **Issue**: There is no way to mark an event I attended, so after a conference, when I am going to watch videos I can't distinguish between those which I wanted to see and those I actually saw.

This is one of my favorites topics which I would love to be resolved in the future. I identified a huge potential for improvement which gives users much more freedom for their use cases. The idea is to grow the binary concept of favorite or not into a tagging schema with a few predefined tags but also allow users to define their own tags. This idea also potentially applies to your use case to filter favorites with alarms as a secondary step. I worked on the concept a bit - see #12. It has not been worked on an implementation, though.

> ... it also makes sense to have a backup plan when a lecture appears boring or streaming breaks ...

Definitely. I also have one or two alternative sessions marked in case something does not work out as expected. In such a case a ranking as conceivable with #12 would be great.

---

Now that commented on your ideas - the question suggests oneself - what happens now? I would love to see some ideas to evolve and to become reality. I am however totally realistic that most of them requires time and involvement of more people. I have been working long enough on this project on my own to know that I am not capable of doing everything myself. Speaking of, if you or somebody you know is interested to join the project (there is more then programming) I would be delighted to guide you and collaborate with you.

**Comment by steelman** [comment]
You've got most things right. So just to clarify a few things.

> I understand that you need the favorites screen to spot the holes if the conference program is so big that the schedule grid does not help that. Here a zoom feature would be a solution.

I don't think zoom would help. At least not with FOSDEM sized schedule. Scrolling up and down the favorites list of about 20 elements and checking their boundaries, even doing simple math, is way simpler that swiping around a schedule that doesn't fit my 24" desktop monitor. Having favorites displayed as a single column in the schedule might be useful but then overlapping backup sessions wouldn't be easy to display. For me personally a simple list of favorites is enough for now.

> If I understand correctly, you are using alarms as a secondary step to filter your favorites list

Yes, and for notifications too. During an online conference like FOSDEM2021 you don't talk much with other people in hallways, but you take a break sometimes to have some tea in the kitchen.

What seems important is to be able to set alarms for either single session or a group of them holding particular label/tag. E.g. I find favorites, I choose goto sessions, I set alarms for all sessions labeled/tagged goto in one move. Alarms wouldn't be a filtering device anymore but only alarms.

> It might be a good idea to merge the alarms with the favorites screen.

Provided that there is still a way to show only "alarmed" sessions. I used the alarms view a few times too.

> One challenge here would be to come up with a good design to fit in all information for a session.

If I were to suggest anything, a bell icon in the upper right corner, and a bunch of colorful lables/tags either below or above the start-time/duration/(end-time/)room [line](https://github.com/EventFahrplan/EventFahrplan/issues/231#issuecomment-776303860). 

> The idea is to grow the binary concept of favorite or not into a tagging schema

Definitely.

-----
I am glad I could share my observations. I wrote some comments for the issues you linked above. I'd love to see these taking shape of new functions in the app, but I know the day is 24 hours long. I don't write for Android. When I feel it is the time to learn it, I am sure this app will be at the top of my list.

**Comment by johnjohndoe** [comment]
Thank you for the clarifications and your further suggestions.

> What seems important is to be able to set alarms for either single session or a group of them holding particular label/tag. E.g. I find favorites, I choose goto sessions, I set alarms for all sessions labeled/tagged goto in one move. Alarms wouldn't be a filtering device anymore but only alarms.

Interesting. I don't remember that **multi-select** was ever mentioned as a feature request.
